io/src/main/scala/sbt/io/IO.scala

```diff
@@ -551,17 +551,23 @@ object IO {
     deleteEmpty(parents(files.toSet))
   }
 
-  /** Deletes `file`, recursively if it is a directory. */
+  /**
+   * Deletes `file`, recursively if it is a directory. Note that this method may silently fail to
+   * delete the file (or directory) if any errors occur.
+   */
   def delete(file: File): Unit = Retry {
     try {
       FileTreeView.default.list(file.toPath).foreach {
         case (dir, attrs) if attrs.isDirectory => delete(dir.toFile)
-        case (f, _)                            => Files.deleteIfExists(f)
+        case (f, _) =>
+          try Files.deleteIfExists(f)
+          catch { case _: IOException => }
```

Comment:
> The legacy behavior of IO.delete is that it leaves behind any files in
the tree that it is unable to delete. This is because File.delete only
throws if an AccessDeniedException occurs. On windows, it is not unusual
for deletions to fail because a jar or class file is in use somewhere.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#delete()

ok. It's yuck, but it makes sense.